src/ParametersHandler/tests/ParametersHandlerTest.cpp

```diff
@@ -14,61 +14,224 @@
+                      const GenericContainer::Vector<const std::string>& parameter) final
+    {
+        return setParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName,
+                      GenericContainer::Vector<int>& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName,
+                      GenericContainer::Vector<double>& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName,
+                      GenericContainer::Vector<std::string>& parameter) const final
     {
-        m_map[name] = std::make_any<shared_ptr>(newGroup);
+        return getParameterPrivate(parameterName, parameter);
     }
 
+public:
+    BasicImplementation(const std::unordered_map<std::string, std::any>& map)
+        : m_map{map}
+    {
+    }
+
+    BasicImplementation() = default;
+
+    bool getParameter(const std::string& parameterName, int& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName, double& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName, std::string& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName, bool& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    bool getParameter(const std::string& parameterName, std::vector<bool>& parameter) const final
+    {
+        return getParameterPrivate(parameterName, parameter);
+    }
+
+    void setParameter(const std::string& parameterName, const int& parameter) final
+    {
+        return setParameterPrivate(parameterName, parameter);
+    }
+
+    void setParameter(const std::string& parameterName, const double& parameter) final
+    {
+        return setParameterPrivate(parameterName, parameter);
+    }
+
+    void setParameter(const std::string& parameterName, const char* parameter) final
+    {
+        return setParameterPrivate(parameterName, std::string(parameter));
+    }
+
+    void setParameter(const std::string& parameterName, const std::string& parameter) final
+    {
+        return setParameterPrivate(parameterName, parameter);
+    }
+
+    void setParameter(const std::string& parameterName, const bool& parameter) final
+    {
+        return setParameterPrivate(parameterName, parameter);
+    }
+
+    void setParameter(const std::string& parameterName, const std::vector<bool>& parameter) final
+    {
+        return setParameterPrivate(parameterName, parameter);
+    }
+
+    bool setGroup(const std::string& name, shared_ptr newGroup)
```

Comment:
I have the feeling it is better to specify that it is taken from ``IParametersHandler``, just to be sure.
```suggestion
    bool setGroup(const std::string& name, IParametersHandler::shared_ptr newGroup)
```

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/28/commits/3ee5e59c0906992c465d17d1afb8a451e6797783